crates/sargon/src/profile/mfa/security_structures/matrices/matrix_of_factor_instances.rs

```diff
@@ -105,12 +173,17 @@ impl MnemonicWithPassphrase {
 impl MatrixOfFactorInstances {
     fn sample_from_matrix_of_sources(
         matrix_of_sources: MatrixOfFactorSources,
+        entity_kind: CAP26EntityKind,
```

Comment:
we now have stricter validation so that we do not accidentally use Account factorinstances for Persona or vice versa.